scripts/RELEASING.md

```diff
@@ -42,7 +42,7 @@ A draft PR can be created, but before releasing consider the following:
 5. [Publish](#publishing-crates) to crates.io using the version as of the release commit
 6. Post to [#otel-rust](https://cloud-native.slack.com/archives/C03GDP0H023) on CNCF Slack.
 
-[Publish.sh](./scripts/publish.sh) may be used to automate steps 3 and 5.
+[Publish.sh](./publish.sh) may be used to automate steps 3 and 5.
```

Comment:
why is RELEASING.md moved under scripts? It is not really a script, and root level actually seemed fine.

Reply:
just looking for a way to hide it from top level directory, to keep the top level directory leaner. (this is not meant for anyone but maintainers to see)
I can move it anywhere else, if there are good alternatives!

Reply:
I don't see any other place to hide it. But I see most of the other languages otel-go, `otel-cpp, otel-java, otel-python have it a top level. Though not that a big issue either, should be fine :)

Reply:
yes. it is immaterial where it is kept. I just prefer to keep top-level leaner and reserve it for things that must absolutely in the root like the toml file etc!